pkg/controller/autoops/secret.go

```diff
@@ -0,0 +1,114 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License 2.0;
+// you may not use this file except in compliance with the Elastic License 2.0.
+
+package autoops
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	autoopsv1alpha1 "github.com/elastic/cloud-on-k8s/v3/pkg/apis/autoops/v1alpha1"
+	esv1 "github.com/elastic/cloud-on-k8s/v3/pkg/apis/elasticsearch/v1"
+	"github.com/elastic/cloud-on-k8s/v3/pkg/controller/common/certificates"
+	"github.com/elastic/cloud-on-k8s/v3/pkg/controller/common/metadata"
+	"github.com/elastic/cloud-on-k8s/v3/pkg/controller/common/reconciler"
+	"github.com/elastic/cloud-on-k8s/v3/pkg/utils/k8s"
+	ulog "github.com/elastic/cloud-on-k8s/v3/pkg/utils/log"
+	"github.com/elastic/cloud-on-k8s/v3/pkg/utils/maps"
+)
+
+const (
+	// autoOpsESCASecretSuffix is the suffix for CA secrets created for each ES instance
+	autoOpsESCASecretSuffix = "autoops-es-ca" //nolint:gosec
+)
+
+// reconcileAutoOpsESCASecret reconciles the Secret containing the CA certificate
+// for a specific Elasticsearch cluster, copying it from the ES instance's http-ca-internal secret.
+func reconcileAutoOpsESCASecret(
+	ctx context.Context,
+	c k8s.Client,
+	policy autoopsv1alpha1.AutoOpsAgentPolicy,
+	es esv1.Elasticsearch,
+) error {
+	log := ulog.FromContext(ctx)
+	log.V(1).Info("Reconciling AutoOps ES CA secret", "namespace", policy.Namespace, "es_namespace", es.Namespace, "es_name", es.Name)
+
+	if es.Status.Phase != esv1.ElasticsearchReadyPhase {
+		log.V(1).Info("Skipping ES cluster that is not ready", "namespace", es.Namespace, "name", es.Name)
+		return nil
+	}
+
+	sourceSecretKey := types.NamespacedName{
+		Namespace: es.Namespace,
+		Name:      certificates.CAInternalSecretName(esv1.ESNamer, es.Name, certificates.HTTPCAType),
+	}
+	var sourceSecret corev1.Secret
+	if err := c.Get(ctx, sourceSecretKey, &sourceSecret); err != nil {
+		if apierrors.IsNotFound(err) {
+			log.V(1).Info("ES http-ca-internal secret not found, skipping", "namespace", es.Namespace, "name", es.Name)
+			return nil
+		}
+		return fmt.Errorf("while retrieving http-ca-internal secret for ES cluster %s/%s: %w", es.Namespace, es.Name, err)
+	}
```

Comment:
IIUC the code [here](https://github.com/elastic/cloud-on-k8s/blob/4110a014ec2a24a69a05d6a25997774421456da9/pkg/controller/common/certificates/reconcile.go#L101) correctly the CA internal can be changed, thus, do we need to add this secret to the dynamic watches?